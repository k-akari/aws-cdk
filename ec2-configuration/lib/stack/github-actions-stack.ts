import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { IamRole } from '../resource/iam-role';

export class GithubActionsStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    new IamRole().createGithubRole(this);
  }
}
