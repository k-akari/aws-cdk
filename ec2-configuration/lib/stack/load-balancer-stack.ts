import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc, CfnSecurityGroup } from 'aws-cdk-lib/aws-ec2';
import { CfnInstance } from 'aws-cdk-lib/aws-ec2';
import { Elb } from '../resource/elb';
import { Subnet } from '../resource/subnet';

export class LoadBalancerStack extends Stack {
  constructor(scope: Construct, id: string, vpc: Vpc, subnet: Subnet, sg: CfnSecurityGroup, ec2Instance: CfnInstance, props?: StackProps) {
    super(scope, id, props);
   
    new Elb(this, vpc, subnet, sg, ec2Instance);
  }
}
