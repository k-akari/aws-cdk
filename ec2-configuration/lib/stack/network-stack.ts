import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { VPC } from '../resource/vpc';
import { Subnet } from '../resource/subnet';
import { InternetGateway } from '../resource/internet-gateway';
import { RouteTable } from '../resource/route-table';
import { NetworkAcl } from '../resource/network-acl';

export class NetworkStack extends Stack {
  public readonly vpc: Vpc;
  public readonly subnet: Subnet;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    const vpc = new VPC(this);
    this.vpc = vpc.vpc;
    this.subnet = new Subnet(this, this.vpc);
    const internetGateway = new InternetGateway(this, this.vpc);
    new RouteTable(this, this.vpc, this.subnet, internetGateway.igw);
    new NetworkAcl(this, this.vpc, this.subnet);
  }
}
