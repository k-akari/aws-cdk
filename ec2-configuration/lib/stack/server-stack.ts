import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { Ec2 } from '../resource/ec2';
import { Subnet } from '../resource/subnet';
import { SecurityGroup } from '../resource/security-group';
import { IamRole } from '../resource/iam-role';
import { ElasticIp } from '../resource/elastic-ip';
import { Elb } from '../resource/elb';

export class ServerStack extends Stack {
  constructor(scope: Construct, id: string, vpc: Vpc, subnet: Subnet, props?: StackProps) {
    super(scope, id, props);

    const sg = new SecurityGroup(this, vpc);

    const instanceProfile = new IamRole().createEc2InstanceProfile(this);
    const ec2Instance = new Ec2().createEc2Instance(this, subnet.public1a, instanceProfile, sg.ec2);
    new ElasticIp().associateElasticIp(this, ec2Instance, 'ec2-1a');

    new Elb(this, vpc, [subnet.public1a, subnet.public1c], sg.alb, ec2Instance);
  }
}
