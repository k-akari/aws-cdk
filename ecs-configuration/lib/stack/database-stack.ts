import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Subnet } from '../resource/subnet';
import { Rds } from '../resource/rds';
import { SecurityGroup } from '../resource/security-group';
import { SecretsManager } from '../resource/secrets-manager';

export class DatabaseStack extends Stack {
  constructor(scope: Construct, id: string, subnet: Subnet, sg: SecurityGroup, ssm: SecretsManager, props?: StackProps) {
    super(scope, id, props);
   
    new Rds(this, subnet, sg, ssm);
  }
}
