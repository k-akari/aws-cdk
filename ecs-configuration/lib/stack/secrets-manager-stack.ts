import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { SecretsManager } from '../resource/secrets-manager';

export class SecretsManagerStack extends Stack {
  public readonly ssm: SecretsManager

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    this.ssm = new SecretsManager(this);
  }
}
