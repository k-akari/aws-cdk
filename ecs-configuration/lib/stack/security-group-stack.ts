import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { SecurityGroup } from '../resource/security-group';

export class SecurityGroupStack extends Stack {
  public readonly sg: SecurityGroup;

  constructor(scope: Construct, id: string, vpc: Vpc, props?: StackProps) {
    super(scope, id, props);

    this.sg = new SecurityGroup(this, vpc);
  }
}
