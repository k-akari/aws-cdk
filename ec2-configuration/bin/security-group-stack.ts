#!/usr/bin/env node
import 'source-map-support/register';
import * as cdk from 'aws-cdk-lib';
import { NetworkStack } from '../lib/stack/network-stack';
import { SecurityGroupStack } from '../lib/stack/security-group-stack';

const app = new cdk.App();
const networkStack = new NetworkStack(app, 'NetworkStack', {});
new SecurityGroupStack(app, 'SecurityGroupStack', networkStack.vpc);