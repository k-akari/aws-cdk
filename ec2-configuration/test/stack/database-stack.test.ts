import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import * as Network from '../../lib/stack/network-stack';
import * as SecurityGroup from '../../lib/stack/security-group-stack';
import * as SecretsManager from '../../lib/stack/secrets-manager-stack';
import * as Database from '../../lib/stack/database-stack';

test('DatabaseStackSnapshot', () => {
  const serviceName = 'service';
  const envType = 'test';
  const app = new App({
    context: {
      'serviceName': serviceName,
      'envType': envType
    }
  });
  const networkStack = new Network.NetworkStack(app, 'NetworkStack');
  const securityGroupStack = new SecurityGroup.SecurityGroupStack(app, 'SecurityGroupStack', networkStack.vpc);
  const secretsManagerStack = new SecretsManager.SecretsManagerStack(app, 'SecretsManagerStack');
  const databaseStack = new Database.DatabaseStack(app, 'DatabaseStack', networkStack.subnet, securityGroupStack.sg, secretsManagerStack.ssm);
  const template = Template.fromStack(databaseStack);

  expect(template).toMatchSnapshot();
});
