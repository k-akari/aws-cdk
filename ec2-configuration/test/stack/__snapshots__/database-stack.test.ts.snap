// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatabaseStackSnapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "DbClusterParameterGroup": Object {
      "Properties": Object {
        "Description": "Cluster Parameter Group for RDS",
        "Family": "aurora-mysql5.7",
        "Parameters": Object {
          "character_set_server": "utf8mb4",
          "collation_server": "utf8mb4_unicode_ci",
          "long_query_time": 1,
          "server_audit_logging": 1,
          "slow_query_log": 1,
          "time_zone": "Asia/Tokyo",
        },
      },
      "Type": "AWS::RDS::DBClusterParameterGroup",
    },
    "ParameterGroupRds": Object {
      "Properties": Object {
        "Description": "Parameter Group for RDS",
        "Family": "aurora-mysql5.7",
      },
      "Type": "AWS::RDS::DBParameterGroup",
    },
    "RdsDbCluster": Object {
      "Properties": Object {
        "BackupRetentionPeriod": 35,
        "DBClusterIdentifier": "service-test-rds-cluster",
        "DBClusterParameterGroupName": Object {
          "Ref": "DbClusterParameterGroup",
        },
        "DBSubnetGroupName": Object {
          "Ref": "SubnetGroupRds",
        },
        "DatabaseName": "planning-poker",
        "EnableCloudwatchLogsExports": Array [
          "general",
          "error",
          "slowquery",
          "audit",
        ],
        "Engine": "aurora-mysql",
        "EngineMode": "provisioned",
        "EngineVersion": "5.7.mysql_aurora.2.10.0",
        "MasterUserPassword": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{{resolve:secretsmanager:",
              Object {
                "Fn::ImportValue": "SecretsManagerStack:ExportsOutputRefSecretRdsClusterF5EFE255",
              },
              ":SecretString:MasterUserPassword}}",
            ],
          ],
        },
        "MasterUsername": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{{resolve:secretsmanager:",
              Object {
                "Fn::ImportValue": "SecretsManagerStack:ExportsOutputRefSecretRdsClusterF5EFE255",
              },
              ":SecretString:MasterUsername}}",
            ],
          ],
        },
        "Port": 3306,
        "PreferredBackupWindow": "18:00-19:00",
        "PreferredMaintenanceWindow": "Mon:19:00-Mon:20:00",
        "StorageEncrypted": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-rds-cluster",
          },
        ],
        "VpcSecurityGroupIds": Array [
          Object {
            "Fn::ImportValue": "SecurityGroupStack:ExportsOutputFnGetAttSecurityGroupRdsGroupIdFE781ECB",
          },
        ],
      },
      "Type": "AWS::RDS::DBCluster",
    },
    "RdsDbInstance1a": Object {
      "Properties": Object {
        "AutoMinorVersionUpgrade": false,
        "AvailabilityZone": "ap-northeast-1a",
        "DBClusterIdentifier": Object {
          "Ref": "RdsDbCluster",
        },
        "DBInstanceClass": "db.r5.large",
        "DBInstanceIdentifier": "service-test-rds-instance-1a",
        "DBParameterGroupName": Object {
          "Ref": "ParameterGroupRds",
        },
        "DBSubnetGroupName": Object {
          "Ref": "SubnetGroupRds",
        },
        "EnablePerformanceInsights": true,
        "Engine": "aurora-mysql",
        "MonitoringInterval": 60,
        "PerformanceInsightsRetentionPeriod": 7,
        "PreferredMaintenanceWindow": "Mon:19:00-Mon:19:30",
        "PubliclyAccessible": false,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-rds-instance-1a",
          },
        ],
      },
      "Type": "AWS::RDS::DBInstance",
    },
    "RdsDbInstance1c": Object {
      "Properties": Object {
        "AutoMinorVersionUpgrade": false,
        "AvailabilityZone": "ap-northeast-1c",
        "DBClusterIdentifier": Object {
          "Ref": "RdsDbCluster",
        },
        "DBInstanceClass": "db.r5.large",
        "DBInstanceIdentifier": "service-test-rds-instance-1c",
        "DBParameterGroupName": Object {
          "Ref": "ParameterGroupRds",
        },
        "DBSubnetGroupName": Object {
          "Ref": "SubnetGroupRds",
        },
        "EnablePerformanceInsights": true,
        "Engine": "aurora-mysql",
        "MonitoringInterval": 60,
        "PerformanceInsightsRetentionPeriod": 7,
        "PreferredMaintenanceWindow": "Mon:19:30-Mon:20:00",
        "PubliclyAccessible": false,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-rds-instance-1c",
          },
        ],
      },
      "Type": "AWS::RDS::DBInstance",
    },
    "SubnetGroupRds": Object {
      "Properties": Object {
        "DBSubnetGroupDescription": "Subnet Group for RDS",
        "DBSubnetGroupName": "service-test-subnet-group-rds",
        "SubnetIds": Array [
          Object {
            "Fn::ImportValue": "NetworkStack:ExportsOutputRefSubnetPrivate1aAFF32B55",
          },
          Object {
            "Fn::ImportValue": "NetworkStack:ExportsOutputRefSubnetPrivate1c235F4ED4",
          },
        ],
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-subnet-group-rds",
          },
        ],
      },
      "Type": "AWS::RDS::DBSubnetGroup",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
